src/jit/gentree.cpp

```diff
@@ -505,6 +525,74 @@ bool GenTree::IsNodeProperlySized() const
 
 /*****************************************************************************/
 
+#if MEASURE_NODE_SIZE
+
+void GenTree::DumpNodeSizes(FILE* fp)
+{
+    // Dump the sizes of the various GenTree flavors
+
+#if SMALL_TREE_NODES
+    fprintf(fp, "Small tree node size = %3u bytes\n", TREE_NODE_SZ_SMALL);
+#endif // SMALL_TREE_NODES
+    fprintf(fp, "Large tree node size = %3u bytes\n", TREE_NODE_SZ_LARGE);
+    fprintf(fp, "\n");
+
+#if SMALL_TREE_NODES
+
+    // Verify that node sizes are set kosherly and dump sizes
+    for (unsigned op = GT_NONE+1; op < GT_COUNT; op++)
+    {
+        unsigned    needSize = s_gtTrueSizes[op];
+        unsigned    nodeSize = s_gtNodeSizes[op];
+
+        const char* structNm = OpStructName((genTreeOps)op);
+        const char* operName =       OpName((genTreeOps)op);
+
+        // Have we seen this struct flavor before?
+        for (unsigned mop = GT_NONE+1; mop < op; mop++)
+        {
+            if (strcmp(structNm, OpStructName((genTreeOps)mop)) == 0)
+            {
+                structNm = nullptr;
+                break;
+            }
+        }
+
+        if (structNm != nullptr)
```

Comment:
It's actually questionable whether it's useful to repeat the error since the name is gone; I should not zap the name and instead use a separate boolean for the repeat, so that all (even repeated) erroneous nodes would print both a name and the "error" thing. That's option 3, in other words. Which do you like best, @pgavlin ?